sky/data/csync/redirect-fuse.c

```diff
@@ -0,0 +1,542 @@
+/*
+  FUSE: Filesystem in Userspace
+  Copyright (C) 2001-2007  Miklos Szeredi <miklos@szeredi.hu>
+  Copyright (C) 2011       Sebastian Pipping <sebastian@pipping.org>
+
+  This program can be distributed under the terms of the GNU GPLv2.
+  See the file COPYING.
+*/
+
+/** @file
+ *
+ * This file system redirects the write and read on mountpoint to other
+ * designated directories to work along with sync daemon and cloud storage
+ * mount under sky/data/csync/sky_csync.py. The code is inspired by an example
+ * provided at:
+ * 		https://github.com/libfuse/libfuse/blob/master/example/passthrough.c		
+ * 
+ * Compile with
+ *
+ *     gcc -Wall -D_FILE_OFFSET_BITS=64 redirect-fuse.c set.c `pkg-config fuse3 --cflags --libs` -o redirect-fuse
+ *
+ * ## Source code ##
+ * \include redirect-fuse.c
+ */
+
+
+#define FUSE_USE_VERSION 31
+
+#define _GNU_SOURCE
+
+#ifdef linux
+#endif
+
+#include <fuse.h>
+#include <libgen.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <dirent.h>
+#include <errno.h>
+#include <stdlib.h>
+#ifdef __FreeBSD__
+#endif
+#include <sys/time.h>
+
+#include "set.h"
+
+static int fill_dir_plus = 0;
```

Comment:
the varname is a little bit confusing; cannot understand the usage when first time saw it.